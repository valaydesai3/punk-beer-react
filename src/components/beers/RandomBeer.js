import React, { useEffect, useContext } from 'react';
import Spinner from '../layout/Spinner';
import AnotherBeer from './AnotherBeer';
import RandomBeerNonAlcoholic from './RandomBeerNonAlcoholic';
import PunkBeerContext from '../../context/punkBeerContext';
import NoImage from '../layout/image-not-available.png';

const RandomBeer = () => {
  const punkBeerContext = useContext(PunkBeerContext);

  const { loading, getRandomBeer, randomBeer } = punkBeerContext;

  useEffect(() => {
    getRandomBeer();
    // eslint-disable-next-line
  }, []);
  const {
    name,
    tagline,
    first_brewed,
    description,
    image_url,
    abv
  } = randomBeer;

  if (loading) return <Spinner />;

  return (
    <div className="card" data-testid="randomBeerCard">
      <h4 className="card-title">{name}</h4>
      <div className="card-grid">
        <div className="card-grid-img">
          <img
            src={image_url === null ? NoImage : image_url}
            alt="Beer"
          />
        </div>
        <div className="card-grid-body">
          <p className="card-grid-description">{description}</p>
          <p className="card-grid-tagline">
            {tagline}
            <span>{first_brewed}</span>
            <span>{abv}&#37;</span>
          </p>
        </div>
        <div className="card-grid-buttons">
          <React.Fragment>
            <AnotherBeer data-testid="anotherBeer" />
            <RandomBeerNonAlcoholic />
          </React.Fragment>
        </div>
      </div>
    </div>
  );
};

export default RandomBeer;
