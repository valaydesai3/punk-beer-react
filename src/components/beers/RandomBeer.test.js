// __tests__/fetch.test.js
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import 'jest-dom/extend-expect';
import axiosMock from 'axios';
import PunkBeerContext from '../../context/punkBeerContext';
import RandomBeer from './RandomBeer';

afterEach(cleanup);

test('loads random beer card', async () => {
  const randomBeer = {
    name: 'Chaos Theory',
    tagline: 'Predictably Random IPA.',
    first_brewed: '10/2008',
    description:
      'Chaos Theory is the most under- rated achievement of 20th Century science. This beer can only aspire to parallel to the mathematical use of the word chaos, which is at odds to the common parlance. The purest showcase of the magnificent hop that is Nelson Sauvin; grapefruit, pineapple and caramel sing above the chaos of life.',
    image_url: 'https://images.punkapi.com/v2/124.png',
    abv: 7.1
  };

  axiosMock.get.mockResolvedValueOnce(() => Promise.resolve(randomBeer));

  let getRandomBeer = jest.fn();
  let setLoading = jest.fn();

  const { container, getByTestId } = render(
    <PunkBeerContext.Provider
      value={{
        randomBeer: randomBeer,
        loading: false,
        getRandomBeer,
        setLoading
      }}
    >
      <RandomBeer />
    </PunkBeerContext.Provider>
  );
  expect(getByTestId('randomBeerCard')).toHaveTextContent('Chaos Theory');
  expect(getByTestId('randomBeerCard')).toHaveTextContent(
    'grapefruit, pineapple and caramel'
  );
  expect(getByTestId('randomBeerCard')).toHaveTextContent(
    'Predictably Random IPA.'
  );
  expect(getByTestId('randomBeerCard')).toHaveTextContent('10/2008');
  expect(getByTestId('randomBeerCard')).toHaveTextContent('7.1');

  const buttons = container.getElementsByTagName('<button>');
  for (let btn of buttons) {
    expect(btn).toHaveClass('btn');
  }

  expect(getByTestId('randomBeerCard')).toMatchSnapshot();
});
