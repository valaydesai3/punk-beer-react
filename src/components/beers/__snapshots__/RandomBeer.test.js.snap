// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads random beer card 1`] = `
<div
  class="card"
  data-testid="randomBeerCard"
>
  <h4
    class="card-title"
  >
    Chaos Theory
  </h4>
  <div
    class="card-grid"
  >
    <div
      class="card-grid-img"
    >
      <img
        alt="Beer"
        src="https://images.punkapi.com/v2/124.png"
      />
    </div>
    <div
      class="card-grid-body"
    >
      <p
        class="card-grid-description"
      >
        Chaos Theory is the most under- rated achievement of 20th Century science. This beer can only aspire to parallel to the mathematical use of the word chaos, which is at odds to the common parlance. The purest showcase of the magnificent hop that is Nelson Sauvin; grapefruit, pineapple and caramel sing above the chaos of life.
      </p>
      <p
        class="card-grid-tagline"
      >
        Predictably Random IPA.
        <span>
          10/2008
        </span>
        <span>
          7.1
          %
        </span>
      </p>
    </div>
    <div
      class="card-grid-buttons"
    >
      <div
        data-testid="anotherBeerBtn"
      >
        <button
          class="btn"
        >
          Another Beer
        </button>
      </div>
      <div
        data-testid="nonAlcoBeerBtn"
      >
        <button
          class="btn"
        >
          Rnadom non-alcoholic beer
        </button>
      </div>
    </div>
  </div>
</div>
`;
